llvm/lib/TableGen/Main.cpp

```diff
@@ -118,9 +118,11 @@ template <class StringSetT>
 static Error createDependencyFile(const StringSetT &Dependencies) {
   return createDependencyFileImpl([&](raw_ostream &OS) {
     OS << OutputFilename << ":";
-    for (StringRef Dep : Dependencies) {
-      OS << ' '
-         << (CreateDependencyFilePM ? CreateDependencyFilePM->map(Dep) : Dep);
+    for (StringRef DepRef : Dependencies) {
+      std::string Dep = DepRef.str();
+      if (CreateDependencyFilePM)
+        cantFail(CreateDependencyFilePM->mapInPlace(Dep));
```

Comment:
Why is this `cantFail`? Same for the one below.  We should either report the error or have a comment why it cannot fail when it's not obvious from immediate context.

Reply:
It's because it creates a `PrefixMapper` instance that doesn't error, but I changed it to propagate a possible error, to be safe for future changes.